runners/reference/java/src/main/java/org/apache/beam/runners/reference/JobServicePipelineResult.java

```diff
@@ -103,11 +112,7 @@ public State waitUntilFinish() {
       response = stub.getState(request);
       lastState = getJavaState(response.getState());
     }
-    try {
-      jobService.close();
-    } catch (Exception e) {
-      LOG.warn("Error cleaning up job service", e);
-    }
```

Comment:
The current behavior of connection cleanup is un-intuitive. The connection use to get closed only if the user called waitUntillFinish while if user did not call WUF then the connection is not closed. Hence removing the connection closing here so that the connection is only closed explicitly in close call.

Reply:
I think it would be ok to have a AutoCloseable close method and call it from here to close eagerly when the terminal state has already been reached.

Reply:
Makes sense